src/widget/textbox.rs

```diff
@@ -133,34 +139,54 @@ impl Widget for TextBox {
 
     fn layout(
         &mut self,
-        bc: &BoxConstraints,
-        _children: &[Id],
-        _size: Option<Size>,
-        _ctx: &mut LayoutCtx,
-    ) -> LayoutResult {
-        LayoutResult::Size(bc.constrain((self.width, BOX_HEIGHT)))
+        _layout_ctx: &mut LayoutCtx,
+        _bc: &BoxConstraints,
+        _data: &String,
+        _env: &Env,
+    ) -> Size {
+        Size::new(self.width, BOX_HEIGHT)
     }
 
-    fn mouse(&mut self, event: &MouseEvent, ctx: &mut HandlerCtx) -> bool {
-        if event.count > 0 {
-            ctx.set_focused(true);
-            ctx.invalidate();
-        }
-        true
-    }
-
-    fn key_down(&mut self, event: &KeyEvent, ctx: &mut HandlerCtx) -> bool {
+    fn event(
+        &mut self,
+        event: &Event,
+        ctx: &mut EventCtx,
+        data: &mut String,
+        _env: &Env,
+    ) -> Option<Action> {
         match event {
-            event if event.key_code == KeyCode::Backspace => {
-                self.text.pop();
+            Event::MouseDown(_) => {
+                if ctx.is_hot() {
```

Comment:
I believe with the new focus logic, what you want to do here is just `ctx.request_focus()`.